geode-benchmarks/src/main/java/org/apache/geode/benchmark/tests/PartitionedWithDeltaAndUniqueObjectReferenceBenchmark.java

```diff
@@ -0,0 +1,242 @@
+import org.apache.geode.perftest.PerformanceTest;
+import org.apache.geode.perftest.Task;
+import org.apache.geode.perftest.TestConfig;
+import org.apache.geode.perftest.TestContext;
+import org.apache.geode.perftest.TestRunners;
+
+public class PartitionedWithDeltaAndUniqueObjectReferenceBenchmark implements PerformanceTest {
+
+  private static final String SUBSCRIPTION_POOL = "subscriptionPool";
+  private static final String ID = UUID.randomUUID().toString();
+
+  public PartitionedWithDeltaAndUniqueObjectReferenceBenchmark() {}
+
+  @Test
+  public void run() throws Exception {
+    TestRunners.defaultRunner().runTest(this);
+  }
+
+  @Override
+  public TestConfig configure() {
+    TestConfig config = GeodeBenchmark.createConfig();
+    config.threads(180);
+    config.durationSeconds(300);
+    config.warmupSeconds(30);
+    ClientServerTopology.configure(config);
+    config.before(new CreatePartitionedRegion(), SERVER);
+    config.before(new CreateClientPool(), CLIENT);
+    config.before(new CreateClientProxyRegionWithPool(), CLIENT);
+    config.workload(new PutWithDeltaTask(ID), CLIENT);
+    return config;
+  }
+
+  public static class PutWithDeltaTask extends BenchmarkDriverAdapter implements Serializable {
+
+    private Region<String, Session> region;
+    private final AtomicReference<String> sessionId = new AtomicReference<>(null);
+
+    private final String id;
+
+    private final List<String> existingSessionAttributeNames =
+        new ArrayList<>();
+
+    private final Random random = new Random(System.currentTimeMillis());
+
+    private Collection<Callable<Integer>> workloadTasks;
+
+    public PutWithDeltaTask(String id) {
+      this.id = id;
+    }
+
+    @Override
+    public void setUp(BenchmarkConfiguration cfg) throws Exception {
+      super.setUp(cfg);
+      this.workloadTasks = newSessionWorkloadTasks();
+      ClientCache cache = ClientCacheFactory.getAnyInstance();
+      cache.setCopyOnRead(true);
+      region = cache.getRegion("region");
+      save(Session.create(id));
+      this.sessionId.set(id);
+    }
+
+    @Override
+    public boolean test(Map<Object, Object> ctx) throws Exception {
+      int sessionAttributeCount = runSessionWorkload();
+
+      assert sessionAttributeCount == this.existingSessionAttributeNames.size();
+      Session session = findById(this.sessionId.get());
+
+      return true;
+    }
+
+    private Session findById(String id) {
+      return Optional.ofNullable(this.region.get(id))
+          .map(Session::commit)
+          .map(Session::touch)
+          .orElseThrow(() -> new IllegalStateException(
+              String.format("No Session with ID [%s] was found", id)));
+    }
+
+    private Session save(Session session) {
+      if (session != null && session.hasDelta()) {
+        this.region.put(session.getId(), session);
+        session.commit();
+      }
+
+      return session;
+    }
+
+    private int runSessionWorkload() {
```

Comment:
The workload is fixed and small so this seems like an overly complicated version of:
```
return workload1() + workload2() + workload3();
```
We end up benchmarking the cost of streams and lambdas too.